### PR TITLE
borgmatic-exporter: init at 0.2.5

### LINKED ISSUE #289977: Package request: borgmatic-exporter
**Project description**

Prometheus exporter for Borgmatic

Written in python, depends on unpackaged python packages [pretty-errors](https://pypi.org/project/pretty-errors) and [timy](https://pypi.org/project/timy).

**Metadata**

* homepage URL: 
* source URL: https://github.com/maxim-mityutko/borgmatic-exporter
* license: mit
* platforms: any

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by flandweber**
## Description of changes

https://github.com/maxim-mityutko/borgmatic-exporter

fix https://github.com/NixOS/nixpkgs/issues/289977

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by flandweber** [comment]
@ofborg test prometheus-exporters.borgmatic

**Comment by IldenH** [comment]
Format with nixfmt-rfc-style
`nix run nixpkgs#nixfmt-rfc-style -- the-files.nix`

**Review by IldenH — APPROVED** [review]

File: pkgs/by-name/pr/prometheus-borgmatic-exporter/package.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  lib,
+  fetchFromGitHub,
+  python3Packages,
+  borgmatic,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "prometheus-borgmatic-exporter";
+  version = "0.2.5";
+
+  format = "pyproject";
```
Comment (SuperSandro2000) [inline pkgs/by-name/pr/prometheus-borgmatic-exporter/package.nix:11]:
```suggestion
    pyproject = true;
```

File: pkgs/by-name/pr/prometheus-borgmatic-exporter/package.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  nativeCheckInputs = with python3Packages; [
+    pytestCheckHook
+    pytest-mock
+  ];
+
+  propagatedBuildInputs =
+    [ borgmatic ]
+    ++ (with python3Packages; [
+      poetry-core
```
Comment (SuperSandro2000) [inline pkgs/by-name/pr/prometheus-borgmatic-exporter/package.nix:28]: poetry-core is a build time only dependency

File: nixos/modules/services/monitoring/prometheus/exporters/borgmatic.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
```
Comment (SuperSandro2000) [inline nixos/modules/services/monitoring/prometheus/exporters/borgmatic.nix:9]:
```suggestion
```
we do not want to add any new of those

File: nixos/modules/services/monitoring/prometheus/exporters/borgmatic.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    '';
+  };
+
+  serviceOpts = {
+    serviceConfig = {
+      DynamicUser = false;
+      ProtectSystem = false;
+      ProtectHome = lib.mkForce false;
+      ExecStart = ''
+        ${pkgs.prometheus-borgmatic-exporter}/bin/borgmatic-exporter run \
```
Comment (SuperSandro2000) [inline nixos/modules/services/monitoring/prometheus/exporters/borgmatic.nix:29]:
```suggestion
        ${lib.getExe pkgs.prometheus-borgmatic-exporter} run \
```
and please add meta.mainProgram to the package.

File: nixos/modules/services/monitoring/prometheus/exporters/borgmatic.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+          --port ${toString cfg.port} \
+          --config ${toString cfg.configFile} \
+          ${concatStringsSep " \\\n  " cfg.extraFlags}
```
Comment (SuperSandro2000) [inline nixos/modules/services/monitoring/prometheus/exporters/borgmatic.nix:32]:
```suggestion
          ${lib.concatMapStringsSep " " (f: lib.escapeShellArg f) cfg.extraFlags}
```
lets keep things simple. Also we should escape the shell args

**Comment by flandweber** [comment]
@SuperSandro2000  Thanks a lot for your help and suggestions!

**Pull request merged**